### PR TITLE
NgAlerting: loki & cortex have different prom & ruler endpoint prefixes

### PR DESCRIPTION
**Pull request by domasx2**
I tested and only Loki worked :D need to use different prom endpoint prefixes for loki and for cortex

Long story: 
Cortex prometheus api prefix is `/api/prom` while Loki prometheus api prefix is `/prometheus`. However in grafana prometheus datasources pointing to cortex are already pre-configured with `/api/prom` prefix, so in the backend proxy we should not add any extra prefix. For Loki we still need to add `/prometheus` prefix.

Ruler endpoints are the same for both Cortex and Loki, `/api/prom/rules*`. However because prometheus datasources already preconfigure the `/api/prom` prefix, we must add this prefix only for requests to Loki.

TL;DR: I fixed it

I tested this to work with both Loki and Cortex

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by papagian** [comment]
These prefixes are configurable by the cli:
https://cortexmetrics.io/docs/api/#path-prefixes
so I think that an installation can have for instance a `legacy-http-prefix` other than `/api/prom`.
If that's true, then we need a way to get the custom prefix.

Also, from [this](https://cortexmetrics.io/docs/api/#list-rules), I understand that both `prometheus-http-prefix` and `legacy-http-prefix` are supported, no?

**Comment by domasx2** [comment]
> These prefixes are configurable by the cli:
> https://cortexmetrics.io/docs/api/#path-prefixes
> so I think that an installation can have for instance a `legacy-http-prefix` other than `/api/prom`.
> If that's true, then we need a way to get the custom prefix.
> 
> Also, from [this](https://cortexmetrics.io/docs/api/#list-rules), I understand that both `prometheus-http-prefix` and `legacy-http-prefix` are supported, no?

looks like! I'd suggest we default to what we use in cloud grafana, and allow customizing the prefix via datasource config. wdyt?

**Review by papagian — APPROVED** [review]
Nice work handling this!
Alternatively, we could split [LotexRulerBackend](https://github.com/grafana/alerting-api/blob/03a29a4c4c27986313d674d0c916695219c8d2df/pkg/api/api.go#L36) to two separate loki and cortex backends since we already check the datasource type for forking.
We still need to figure out how to support custom prefixes.
